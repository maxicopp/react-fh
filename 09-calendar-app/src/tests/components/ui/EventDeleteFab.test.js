import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import '@testing-library/jest-dom';
import { EventDeleteFab } from '../../../components/ui/EventDeleteFab';
import { eventStartDelete } from '../../../components/actions/events';

jest.mock('../../../components/actions/events', () => ({
    eventStartDelete: jest.fn()
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {};
const store = mockStore(initState);
store.dispatch = jest.fn();

const wrapper = mount(
    <Provider store={store}>
        <EventDeleteFab />
    </Provider>
)

describe('Test in <EventDeleteFab />', () => {
    test('should be displayed correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should call the eventStartDelete when clicking', () => {
        wrapper.find('button').prop('onClick')();
        expect(eventStartDelete).toHaveBeenCalled();
    });


});
