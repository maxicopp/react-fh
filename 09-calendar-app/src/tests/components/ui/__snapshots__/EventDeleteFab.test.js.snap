// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <EventDeleteFab /> should be displayed correctly 1`] = `
<button
  className="btn btn-danger fab-danger"
  onClick={[Function]}
>
  <i
    className="fas fa-trash"
  />
  <span>
     Borrar evento 
  </span>
</button>
`;
